### PR TITLE
test_result_logger: accept release version numbers

### PR DESCRIPTION
**Pull request by ArchangeGabriel**
When building on release, the output version is e.g. 0.8.0, without dev part.
Thus, this commit makes the dev part optional, allowing it to match both cases.
Fixes GH-420.

### CODE REVIEW

File: whipper/test/test_result_logger.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def testLogger(self):
             actualLines[0],
             re.compile((
                 r'Log created by: whipper '
-                r'[\d]+\.[\d]+.[\d]+\.dev[\w\.\+]+ \(internal logger\)'
+                r'[\d]+\.[\d]+.[\d]+(\.dev[\w\.\+]+)? \(internal logger\)'
```
Comment (ArchangeGabriel) [inline whipper/test/test_result_logger.py]: I’m wondering if the initial rule isn’t missing a `\` before the second `.`…

Comment (Freso) [reply whipper/test/test_result_logger.py]: It is. (But that should be a separate PR.)

**Comment by Freso** [comment]
Ideally there should be at least four test cases, one for each of the situations described at https://github.com/pypa/setuptools_scm/#default-versioning-scheme

File: whipper/test/test_result_logger.py (hunk 1 above)
Comment (JoeLametta) [reply whipper/test/test_result_logger.py]: Addressed it in a separate commit (f740a0ef0d4503b16485b417d1222032a20cd4b6) and rebased this branch on develop.

**Comment by JoeLametta** [comment]
@Freso @ArchangeGabriel I think the following RegEX matches all the (4) cases:

```regex
[\d]+\.[\d]+\.[\d]+(\+d\d{8}|\.dev[\w\.\+]+)? \(internal logger\)
```

**Pull request closed without merging**